src/stampy_nlp/routes.py

```diff
@@ -51,18 +61,22 @@ def wrapped_view(**kwargs):
 def log_queries(f):
     @wraps(f)
     def wrapper(*args, **kwargs):
-        if request.method == 'GET':
-            query = request.args.get('query', DEFAULT_QUERY)
-        elif request.method == 'POST':
-            query = request.form.get('query', DEFAULT_QUERY)
-        else:
-            query = None
-
+        query = get_query(request)
         log_query(request.path, request.method, query)
         return f(*args, **kwargs)
     return wrapper
 
 
+def require_query(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        query = get_query(request)
+        if not (query and query.strip()):
```

Comment:
this will return a 400 error if the `query` param:
* is `None` (potentially could happen if someone sends a POST, but puts the query in the url params)
* is set to an empty string
* is set to whitespace, e.g. `'     \n\n   \t\t      '`